openpype/hosts/blender/scripts/build_workfile.py

```diff
@@ -747,18 +750,59 @@ def build_anim(project_name, asset_name):
                 use_selection=False,
             )
 
-    for obj in bpy.context.scene.objects:
-        if obj.type == "ARMATURE":
-            # Create animation instance
-            variant_name = obj.name[obj.name.find("RIG_") + 4 :].capitalize()
-            bpy.ops.scene.create_openpype_instance(
-                creator_name="CreateAnimation",
-                asset_name=asset_name,
-                subset_name=f"animation{variant_name}",
-                datapath="objects",
-                datablock_name=obj.name,
-                use_selection=False,
+    instances_to_create = {}
+    for container in bpy.context.scene.openpype_containers:
+        container_metadata = container["avalon"]
+        variant_name = container_metadata.get("asset_name")
+        family = container_metadata.get("family")
+        container_datablocks = container.get_datablocks(bpy.types.Object)
+
+        if family == "setdress":
+            instances_to_create[variant_name] = list(
+                container.get_root_outliner_datablocks()
             )
+        else:
+            # Get rigs
+            armature_objects = [
+                o for o in container_datablocks if o.type == "ARMATURE"
+            ]
+            # Get animated objects
+            animated_objects = [
+                o for o in container_datablocks
+                if o.animation_data and o.animation_data.action
+            ]
+
+            if armature_objects or animated_objects:
+                instances_to_create[variant_name] = (
+                    armature_objects + animated_objects
+                )
+
+    # Create instances and add datablocks
+    for variant_name, objects in instances_to_create.items():
+        first_datablock = objects.pop(0)
+        datablock_type = (
+            "collections"
+            if isinstance(first_datablock, bpy.types.Collection)
+            else "objects"
+        )
+        bpy.ops.scene.create_openpype_instance(
+            creator_name="CreateAnimation",
+            asset_name=asset_name,
+            subset_name=f"animation{variant_name}",
+            datapath=datablock_type,
```

Comment:
You better use `BL_TYPE_DATAPATH.get(type(first_datablock))`

Reply:
Also, can't you leave `datapath` and `datablock_name` empty and add to container all of them without pop the first one?

Reply:
I already try without datapath and datablock_name, get this result :
```python
RuntimeError: Error: Python: Traceback (most recent call last):
  File "D:\Kevin\DEV\OpenPype\openpype\hosts\blender\api\ops.py", line 603, in execute
    plugin.process(
  File "D:\Kevin\DEV\OpenPype\openpype\hosts\blender\api\plugin.py", line 378, in wrapper_exec_process
    return process_func(*args, **kwargs)
  File "D:\Kevin\DEV\OpenPype\openpype\hosts\blender\api\plugin.py", line 558, in process
    if op_instance.datablock_refs.get(d.name):
AttributeError: 'NoneType' object has no attribute 'name'
```

Reply:
few changes but can't leave `datapath `and `datablock_name` empty.